pkg/schema/factory.go

```diff
@@ -145,11 +145,11 @@ func (c *Collection) schemasForSubject(access *accesscontrol.AccessSet) (*types.
 		s = s.DeepCopy()
 		attributes.SetAccess(s, verbAccess)
 
-		if s.ResourceFields == nil {
-			s.ResourceFields = make(map[string]schemas.Field)
+		if s.Attributes == nil {
+			s.Attributes = make(map[string]interface{})
 		}
-		s.ResourceFields["resourcePermissions"] = schemas.Field{
-			Type:        "map[json]",
+		s.Attributes["resourcePermissions"] = schemas.Field{
```

Comment:
You stated your intent to make `resourcePermissions` peer with `resourceMethods` and `resourceFields`, but this appears to instead be moving it under `attributes` instead of under `resourcePermissions` - where it clearly did not belong.

Are we just randomly picking a place in the schema for some new data? Where does this actually **BELONG**?

Reply:
Fair point....
I should have been more clear.
This move puts resourcePermissions on the same level as "verbs" under attributes, which I think makes the most sense since its purpose is similar.

Reply:
Has there been any higher-level discussion about where this field belongs? What is consuming this new field? I don't see that it is getting populated anywhere - this sets the type and description but where is it actually used?

Reply:
ok but it sounds like that didn't address where specifically to put the field?

Reply:
I've been in contact with @richard-cox for work on the epic, we've walked through things.  It was entirely my mistake for putting it under resourceFields at first and we didn't catch it until now.  I misunderstood the purpose of them.

Reply:
OK. So stuffing this into the a key under the `Attributes map[string]interface{}` is the correct thing to do here, instead of making it a new peer to the other `Resource*` fields?

https://github.com/rancher/wrangler/blob/a1fbf41856b2cc65bf4f644b2b1a5d4fd88a24be/pkg/schemas/types.go#L10-L12

Reply:
It currently contains other information related to things that the current user can/can't do.  disallowMethods, and verbs are other bits of information that are currently in the attributes field.

Reply:
It looks like other things that interact with Attributes fields are all done with helpers in `attributes.go`:
https://github.com/rancher/steve/blob/ea01a40c8ddfb1d988501aa64c3d19911202d7e7/pkg/attributes/attributes.go#L166-L186

Should this perhaps be done via `SetResourcePermissions`/`ResourcePermissions` helper functions?

Reply:
Sure, I suppose that would be a good way to go. Thanks.

Reply:
Upon further review (and talking with Richard), it sounds like the most proper way forward here is to define a new field in Wrangler that is specific to resource permissions.  While attributes does indeed have some other related information, it sounds like the intended original purpose of the attributes field was more related to CRD columns.  Rather than add more "things" that run contrary to the original purpose, we will take care of this at the wrangler level. 

So this PR will become "remove the manual addition of the schema field and bump wrangler".
Set back to draft for now until the wrangler bump is ready.

Reply:
@brandond Any other thoughts on this?

Reply:
I am 👍🏻 on moving it to a proper field in `schemas.Schema`